internal/overlord/servstate/handlers.go

```diff
@@ -162,7 +168,20 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 			}
 			releasePlan()
 		}
-		return fmt.Errorf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+
+		it := logBuffer.HeadIterator(lastLogLines)
+		defer it.Close()
+		lastLogs, err := ioutil.ReadAll(it)
+		if err != nil {
+			logger.Noticef("could not read service %q logs: %v", req.Name, err)
+		}
+
+		msg := fmt.Sprintf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+		if len(lastLogs) > 0 {
+			msg = fmt.Sprintf("%s - last %d lines of output:\n%s",
```

Comment:
I'm hoping we can get something a bit simpler and less verbose.  How about something like this:
```
cannot start %q: exited quickly with code %d:
    (...)
    <line1>
    <line2>
```
Four spaces at the start, and (...) is only present if there was any output before it. We have logic to do almost exactly this in a function called OutputErr somewhere (likely strutil or osutil).

Also, the timestamp and extra service name on every line is not nice. It makes sense on an aggregated log, but here it seems alien and unpolished. We should be able to produce a cleaner log, either using the same logic used elsewhere (reusing, not copying), or maybe with the knowledge that we have above (the formatting is setup right here).

Happy to talk further.

Reply:
I've fixed this now to simplify the message (we don't actually need the `%q` service name as it's already in the outer message (see updated PR description), but I've made the other changes: indentation by 4 spaces, and the `(...)` truncation marker if it exceeds 20 lines.

I'm not planning to remove the timestamps for now, as we have a separate task on our [TODO list](https://github.com/canonical/pebble#roadmap--todo) to parse/remove timestamps, and we can clean this up then as well.

Reply:
Per discussion, I'm stripping off the timestamp + service name now.